benchmarks/lambda_vs_nested_methods.rb

```diff
@@ -0,0 +1,23 @@
+require 'benchmark/ips'
+
+def nested_methods
+  def some_method
+  end
+end
+
+def lambda_method
+  some_lambda = lambda {|i|}
+end
```

Comment:
These are not equivalent.  The lambda here is a local variable that is only visible in the method.  The nested method is adding the `some_method` method to the object when `nested_methods` is called, which mutates the object and makes the method available outside the scope of `nested_method`.  It also blows away the method cache and is generally not a good technique.